pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -345,6 +346,14 @@ func (rq *Controller) syncResourceQuota(resourceQuota *v1.ResourceQuota) (err er
 	hardResources := quota.ResourceNames(hardLimits)
 	used = quota.Mask(used, hardResources)
 
+	// use the same format in used as in hard to prevent inconsistent unit format switching
+	for resourceName, usedQuantity := range used {
+		hardQuantity := hardLimits[resourceName]
+		if hardQuantity.Format != usedQuantity.Format {
+			used[resourceName] = *resource.NewQuantity(usedQuantity.Value(), hardQuantity.Format)
+		}
+	}
+
```

Comment:
the above change will not be updated unless dirty (see line 368 below)

Reply:
please see https://github.com/kubernetes/kubernetes/pull/102178#discussion_r638554454